File: finalAng/src/app/model/ppdistribution.ts
export class PPDistribution {
  public get mail(): string {
    return this._mail;
  }
  public set mail(value: string) {
    this._mail = value;
  }
  public get telephone2(): string {
    return this._telephone2;
  }
  public set telephone2(value: string) {
    this._telephone2 = value;
  }
  public get telephone1(): string {
    return this._telephone1;
  }
  public set telephone1(value: string) {
    this._telephone1 = value;
  }
  public get fax(): string {
    return this._fax;
  }
  public set fax(value: string) {
    this._fax = value;
  }
  public get dateDernierModification(): Date {
    return this._dateDernierModification;
  }
  public set dateDernierModification(value: Date) {
    this._dateDernierModification = value;
  }
  public get rejet(): number {
    return this._rejet;
  }
  public set rejet(value: number) {
    this._rejet = value;
  }
  public get prenom(): string {
    return this._prenom;
  }
  public set prenom(value: string) {
    this._prenom = value;
  }
  public get nom(): string {
    return this._nom;
  }
  public set nom(value: string) {
    this._nom = value;
  }
  public get civilite(): string {
    return this._civilite;
  }
  public set civilite(value: string) {
    this._civilite = value;
  }
  public get IdPPDistribution(): number {
    return this._IdPPDistribution;
  }
  public set IdPPDistribution(value: number) {
    this._IdPPDistribution = value;
  }
  constructor(
    private _IdPPDistribution: number,
    private _civilite: string,
    private _nom: string,
    private _prenom: string,
    private _telephone1: string,
    private _telephone2: string,
    private _mail: string,
    private _fax: string,
    private _dateDernierModification: Date,
    private _rejet: number
  ) {}
}

File: finalRest/src/main/java/ajc/formation/project/finalRest/entity/PPDistribution.java
package ajc.formation.alten.finalRest.entity;

import java.sql.Date;
import java.util.Set;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonView;

import ajc.formation.alten.finalRest.jsonview.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "PPDISTRIBUTION")
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class PPDistribution {

	@Id
	@EqualsAndHashCode.Include
	@JsonView(Views.Common.class)
	@Column(name="IDPPDISTRIBUTION")
	private Long idPPDistribution;

	@JsonView(Views.Common.class)
	@Enumerated(EnumType.STRING)
	@Column(name = "CIVILITE")
	private Civilite civilite;

	@Column(length = 40, nullable = false,name="NOM")
	@NotBlank
	@JsonView(Views.Common.class)
	private String nom;

	@JsonView(Views.Common.class)
	@Column(length = 40,name="PRENOM")
	private String prenom;

	@JsonView(Views.Common.class)
	@Column(length = 14,name="TELEPHONE1")
	private String telephone1;

	@JsonView(Views.Common.class)
	@Column(length = 14, name="TELEPHONE2")
	private String telephone2;

	@JsonView(Views.Common.class)
	@Column(length = 100, name = "MAIL")
	private String mail;

	@JsonView(Views.Common.class)
	@Column(length = 14,name="FAX")
	private String fax;

	@Column(name="DATEDERNIEREMODIFICATION")
	@JsonView(Views.Common.class)
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	private Date dateDernierModification;

	@JsonView(Views.Common.class)
	@Column(name="REJET")
	private Integer rejet;

	@OneToMany(mappedBy = "responsable")
	private Set<Reseau>  reseaux;

	@OneToMany(mappedBy = "ppDistributionn")
	private Set<Courtier> courtiers;

	@OneToMany(mappedBy = "distributionWithAgence")
	private Set<Agence> agenceWithdistributions;
}
